Intermittant CI Failure

The following test intermittently fails in CI; re-running usually clears the error. Any thoughts as to what we can do to improve this @jrwrigh?
```
        TEST fluids-py-smartsim_regression_framework
13:03:30 noether SmartSim[993423] ERROR Orchestrator failed during startup See /var/lib/gitlab-runner/builds/N8GCsqus/2/libceed/libCEED/tests/test_dir/test/database for details
Traceback (most recent call last):
  File "/var/lib/gitlab-runner/builds/N8GCsqus/2/libceed/libCEED/tests/junit.py", line 197, in <module>
    test_framework.setup()
  File "/var/lib/gitlab-runner/builds/N8GCsqus/2/libceed/libCEED/examples/fluids/smartsim_regression_framework.py", line 51, in setup
    self.exp.start(self.database)
  File "/home/jawr8143/SmartSimTestingSoftware/smartsim/smartsim/experiment.py", line 191, in start
    self._control.start(
  File "/home/jawr8143/SmartSimTestingSoftware/smartsim/smartsim/_core/control/controller.py", line 95, in start
    self._launch(manifest)
  File "/home/jawr8143/SmartSimTestingSoftware/smartsim/smartsim/_core/control/controller.py", line 323, in _launch
    self._launch_orchestrator(orchestrator)
  File "/home/jawr8143/SmartSimTestingSoftware/smartsim/smartsim/_core/control/controller.py", line 384, in _launch_orchestrator
    self._orchestrator_launch_wait(orchestrator)
  File "/home/jawr8143/SmartSimTestingSoftware/smartsim/smartsim/_core/control/controller.py", line 584, in _orchestrator_launch_wait
    raise SmartSimError(msg)
smartsim.error.errors.SmartSimError: Orchestrator failed during startup See /var/lib/gitlab-runner/builds/N8GCsqus/2/libceed/libCEED/tests/test_dir/test/database for details
make: *** [Makefile:678: junit-fluids-py-smartsim_regression_framework] Error 1
```

--- a/examples/fluids/smartsim_regression_framework.py
+++ b/examples/fluids/smartsim_regression_framework.py
@@ -13,6 +13,7 @@ from typing import Tuple
 import os
 import shutil
 import logging
+import socket
 
 # autopep8 off
 sys.path.insert(0, (Path(__file__).parents[3] / "tests/junit-xml").as_posix())
@@ -21,6 +22,14 @@ sys.path.insert(0, (Path(__file__).parents[3] / "tests/junit-xml").as_posix())
 logging.disable(logging.WARNING)
 
 fluids_example_dir = Path(__file__).parent.absolute()
+
+
+def getOpenSocket():
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.bind(('', 0))
+    addr = s.getsockname()
+    s.close()
+    return addr[1]
 
 
 class NoError(Exception):
@@ -44,7 +53,7 @@ class SmartSimTest(object):
         self.directory_path.mkdir()
         os.chdir(self.directory_path)
 
-        PORT = 6780
+        PORT = getOpenSocket()
         self.exp = Experiment("test", launcher="local")
         self.database = self.exp.create_database(port=PORT, batch=False, interface="lo")
         self.exp.generate(self.database)

